### PR TITLE
My personal module for my rpg Jorik

### PR DESCRIPTION
**Pull request by T-R-I-X**
This should help alot of people looking for a good module that can last through alot of Roblox api changes. I have rewritten the old module to work with Robloxs' current api.

### CODE REVIEW

File: modules/EasyNetwork.lua [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+end
+
+-- .. Waits for a child, more expensive then using object:WaitForChild
+local function WaitForChild(parent, name, timeLimit)	
+	local child = parent:FindFirstChild(name)
+	local startTime = tick()
+	local warned = false
+	
+	-- ## Loops until child is 
+	while not child do
```
Comment (RigidStudios) [inline modules/EasyNetwork.lua:69]: Check if child first

File: modules/EasyNetwork.lua [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+local osTime, newInstance, traceback = os.time, Instance.new, debug.traceback
+
+-- Values
+local maxWait = 1
+local defaultCache = 10
+
+-------- Private Functions --------
+
+-- .. Creates a instance
+local function Create(classType, properties)
```
Comment (RigidStudios) [inline modules/EasyNetwork.lua]: Really quite bad for memory, instance them normally.

File: modules/EasyNetwork.lua [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+	local object = newInstance(classType)
+	
+	for property, propertyValue in next, properties do
+		object[property] = propertyValue
+	end
+	
+	return object
+end
+
+-- .. Waits for a child, more expensive then using object:WaitForChild
```
Comment (RigidStudios) [inline modules/EasyNetwork.lua:61]: Use the waitforchild method then lmao

File: modules/EasyNetwork.lua [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+function remoteEvent:SendToPlayers(playerList, ...)
+	assert(server, "SendToPlayers should be called from the Server side")
+
+	for a = 1, #playerList do
+		self.Instance:FireClient(playerList[a], ...)
+	end
+end
+
+-- .. Fires the client with the arguments difined
+function remoteEvent:SendToPlayer(player, ...)
```
Comment (RigidStudios) [inline modules/EasyNetwork.lua:272]: vararg Still a possibility.

File: modules/EasyNetwork.lua (hunk 3 above)
Comment (T-R-I-X) [reply modules/EasyNetwork.lua:61]: It wouldn't matter

**Pull request closed without merging**